src/main/java/ch/njol/skript/events/EvtItem.java

```diff
@@ -74,10 +75,20 @@ public class EvtItem extends SkriptEvent {
 						"\t\tset item of event-dropped item to a diamond")
 				.since("<i>unknown</i> (before 2.1), INSERT VERSION (entity)");
 		if (hasPrepareCraftEvent) { // Must be loaded before CraftItemEvent
-			Skript.registerEvent("Prepare Craft", EvtItem.class, PrepareItemCraftEvent.class, "[player] (preparing|beginning) craft[ing] [[of] %-itemtypes%]")
+			Class<? extends Event>[] events;
+			if (hasPlayerStonecutterRecipeSelectEvent) {
+				events = CollectionUtils.array(PrepareItemCraftEvent.class, PlayerStonecutterRecipeSelectEvent.class);
+			} else {
+				events = CollectionUtils.array(PrepareItemCraftEvent.class);
+			}
+			Skript.registerEvent("Prepare Craft", EvtItem.class, events, "[player] (preparing|beginning) craft[ing] [[of] %-itemtypes%]")
```

Comment:
I don't think this syntax reflects the stone cutter recipe event. It should have it's own syntax to relate more to the event name.

You could register two registerEvent methods one with the new event and the new syntax and the other method call with the prepare item craft event and the old syntax with both referencing the EvtItem class.